src/components/molecules/modal/index.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import { ReactComponent as CloseIcon } from "../../../assets/svg/2.0/20px/x.svg"
+import * as Dialog from '@radix-ui/react-dialog'
+
+const Overlay = ({children, ...rest}) => {
+  return (<Dialog.Overlay className='fixed bg-grey-90/40 grid top-0 left-0 right-0 bottom-0 place-items-center overflow-y-auto'>
+    {children}
+  </Dialog.Overlay>)
+}
+
+const Content = ({children, ...rest}) => {
+  return (<Dialog.Content className='bg-grey-0 min-w-modal rounded'>
+    {children}
+  </Dialog.Content>)
+}
+
+const addProp = (children, prop) => {
+  return !Array.isArray(children) ? React.cloneElement(children, prop) : 
+    children.map(child => React.cloneElement(child, prop))
+}
```

Comment:
nit: Use `React.Children.map` from the React API to handle array, single element, undefined and null more elegantly.  

```
return React.Children.map(
    children, 
    (child, I) => React.cloneElement(child, prop)
)
```

Reply:
This seems like a much better way to do it 😄  was unaware of the `React.Children.map` function and it is implemented now. 

I also added proptypes and did some more cleanup